app/models/file_depot.rb

```diff
@@ -33,7 +33,7 @@ def upload_file(file)
     @file = file
   end
 
-  def merged_uri(_uri, _api_port)
+  def merged_uri(uri, _api_port)
```

Comment:
So, this being the only method change, and after doing a bit of "`git` spelunking" I have noticed that this method and single method that calls it (`MiqSchedule#verify_file_depot`), has had a decent amount of churn around them:

- https://github.com/ManageIQ/manageiq/pull/18131
- https://github.com/ManageIQ/manageiq/pull/17967

My main concern isn't really with the change, but that how we are testing it and that it doesn't seem to expose the issue that has been coming up in previous commit/PR/spec changes.

What I mean by that is `MiqSchedule#verify_file_depot` doesn't seem to have ever failed, or tests anything besides a single `FTP` example.  But a big part of the code that would even call on this method is based on if an `api_port` param is passed in to `verify_file_depot`:

https://github.com/ManageIQ/manageiq/blob/e3e7c38b24a445f50ded7bf15438fadd0da9294a/app/models/miq_schedule.rb#L345

If it is blank (and it should be for almost every `FileDepot` type besides `swift`), then this method should have never been called in the first place, right?

* * *

I guess what I am confused about is how we have a non-blank `api_port` in the first place, when only `swift` should be passing this around in the first place.  It seems like this would explain why it was working in the first place, but why it is now failing (somewhat... I think).

Seems like adding tests around `verify_file_depot` should be considered, and maybe as the `TODO` for this method suggests, adding it to the UI specs as well?

Reply:
The UI actually defaults the port to 5000 if its not set, so at least for these calls to verify_file_depot() merged_uri will always be called.  That is ok since except for Swift, we will simply return the uri (first) argument.

Reply:
@NickLaMuro I've added a couple of more tests in miq_schedule_spec.rb to handle the verify_file_depot cases of a null and non-null swift port and what it does with regard to calling merged_uri.  This should do what you want, hopefully.